pkg/ovs/daemonset.go

```diff
@@ -0,0 +1,205 @@
+	envVars["OvnEncapType"] = env.SetValue(instance.Spec.ExternalIDS.OvnEncapType)
+	envVars["OvnEncapIP"] = env.SetValue(instance.Spec.ExternalIDS.OvnEncapIP)
+	envVars["EnableChassisAsGateway"] = env.SetValue(fmt.Sprintf("%t", instance.Spec.ExternalIDS.EnableChassisAsGateway))
+	envVars["PhysicalNetworks"] = env.SetValue(getPhysicalNetworks(instance))
+
+	daemonset := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceName,
+			Namespace: instance.Namespace,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+					Annotations: map[string]string{
+						"k8s.v1.cni.cncf.io/networks": getNetworksList(instance),
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: ServiceAccountName,
+					Containers: []corev1.Container{
+						// ovsdb-server container
+						{
+							Name: ServiceName + "db-server",
+							Command: []string{
+								"/usr/bin/start-ovs",
+							},
+							Args: []string{
+								"ovsdb-server",
+							},
+							Image: instance.Spec.OvsContainerImage,
+							SecurityContext: &corev1.SecurityContext{
+								Capabilities: &corev1.Capabilities{
+									Add:  []corev1.Capability{"NET_ADMIN", "SYS_ADMIN", "SYS_NICE"},
+									Drop: []corev1.Capability{},
+								},
+								RunAsUser:  &runAsUser,
+								Privileged: &privileged,
+							},
+							Env:           env.MergeEnvs([]corev1.EnvVar{}, envVars),
+							VolumeMounts:  GetOvsDbVolumeMounts(),
+							Resources:     instance.Spec.Resources,
+							LivenessProbe: ovsDbLivenessProbe,
+						}, {
+							// ovs-vswitchd container
+							Name: ServiceName + "-vswitchd",
+							Command: []string{
+								"/usr/bin/start-ovs",
+							},
+							Args: []string{
+								"ovs-vswitchd",
+							},
+							Image: instance.Spec.OvsContainerImage,
+							SecurityContext: &corev1.SecurityContext{
+								Capabilities: &corev1.Capabilities{
+									Add:  []corev1.Capability{"NET_ADMIN", "SYS_ADMIN", "SYS_NICE"},
+									Drop: []corev1.Capability{},
+								},
+								RunAsUser:  &runAsUser,
+								Privileged: &privileged,
+							},
+							Env:           env.MergeEnvs([]corev1.EnvVar{}, envVars),
+							VolumeMounts:  GetVswitchdVolumeMounts(),
+							Resources:     instance.Spec.Resources,
+							LivenessProbe: ovsVswitchdLivenessProbe,
+						}, {
+							// ovn-controller container
+							Name: OvnControllerServiceName,
+							Command: []string{
+								"/bin/bash", "-c",
+							},
+							Args: []string{
+								// First configure external ids and then start ovn controller
+								"/usr/local/bin/container-scripts/init.sh && /usr/bin/ovn-controller --pidfile --log-file unix:/run/openvswitch/db.sock",
+							},
+							Image: instance.Spec.OvnContainerImage,
+							SecurityContext: &corev1.SecurityContext{
+								Capabilities: &corev1.Capabilities{
```

Comment:
something to check for later if we really need these security contexts for ovn-controller.

Reply:
I added TODO